test/src/test/resources/fixtures/new hydration/new batching/new-batching-conditional-hydration-with-list-condition-field.yml

```diff
@@ -0,0 +1,121 @@
+name: "new batching conditional hydration with list condition field"
+enabled: true
+overallSchema:
+  service2: |
+    type Query {
+      barsById(ids: [ID]): [Bar]
+    }
+    type Bar {
+      id: ID
+      name: String
+    }
+  service1: |
+    type Query {
+      foo: Foo
+    }
+    type Foo {
+      id: ID
+      barIds: [ID] @hidden
+      bar: [Bar] 
+        @hydrated(
+          service: "service2"
+          field: "barsById"
+          arguments: [
+            {
+              name: "ids"
+              value: "$source.barIds"
+            }
+          ]
+          when: {
+            result: {
+              sourceField: "barIds"
+              predicate: { equals: "barId2" }
+            }
+          }
+        )
+    }
+underlyingSchema:
+  service2: |
+    type Bar {
+      id: ID
+      name: String
+    }
+
+    type Query {
+      barsById(ids: [ID]): [Bar]
+    }
+  service1: |
+    type Foo {
+      barIds: [ID]
+      id: ID
+    }
+
+    type Query {
+      foo: Foo
+    }
+query: |
+  query {
+    foo {
+      bar {
+          name
+      }
+    }
+  }
+variables: { }
+serviceCalls:
+  - serviceName: "service1"
+    request:
+      query: |
+        query {
+          foo {
+            __typename__batch_hydration__bar: __typename
+            batch_hydration__bar__barIds: barIds
+            batch_hydration__bar__barIds: barIds
+          }
+        }
+      variables: { }
+    # language=JSON
+    response: |-
+      {
+        "data": {
+          "foo": {
+            "__typename__batch_hydration__bar": "Foo",
+            "batch_hydration__bar__barIds": ["barId1", "barId2", "barId3"]
+          }
+        },
+        "extensions": {}
+      }
+  - serviceName: "service2"
+    request:
+      query: |
+        query {
+          barsById(ids: ["barId2"]) {
+            batch_hydration__bar__id: id
+              name
+          }
+        }
+      variables: { }
+    # language=JSON
+    response: |-
+      {
+        "data": {
+          "barsById": [{
+            "batch_hydration__bar__id": "barId1",
+            "name": "Bar1"
```

Comment:
I was going to say remove this because `barId1` is absent in the query, but in theory the server could reply with this and it's good to see we never put it into the response.